### PR TITLE
Add docs for exchange materialization

### PR DESCRIPTION
**Pull request by wenleix**
Test plan - build the doc locally

Screen Shot 2020-09-19 at 3 54 07 PM

-----------------------

Screen Shot 2020-09-19 at 3 54 15 PM

-----------------------

Screen Shot 2020-09-19 at 3 54 22 PM

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by arhimondr — APPROVED** [review]

**Comment by arhimondr** [comment]
@wenleix Is grouped execution enabled by default?

**Comment by wenleix** [comment]
@arhimondr 

> @wenleix Is grouped execution enabled by default?

Yes!

https://github.com/prestodb/presto/blob/79c3a3f26e4e771eed3f568d0ead36f41131d72c/presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java#L70-L74

File: presto-docs/src/main/sphinx/admin/exchange-materialization.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    GROUP BY custkey
+
+
+The following figure demonstrates how this query executes in Presto classic mode:
+
+.. figure:: ../images/rpc_shuffle_execution.png
+   :align: center
+
+With exchange materialization, the intermediate shuffle data is written to disk (currently,
+it is always a temporary Hive bucketed table). This opens opportunity for flexible scheduling policies
```
Comment (aweisberg) [inline presto-docs/src/main/sphinx/admin/exchange-materialization.rst:32]:
```suggestion
it is always a temporary Hive bucketed table). This creates an opportunity for flexible scheduling policies
```

File: presto-docs/src/main/sphinx/admin/exchange-materialization.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+on the aggregation side, as only a subset of aggregation data needs to be held in memory at the
+same time -- this execution strategy is called "grouped execution" in Presto.
+
+.. figure:: ../images/materialized_shuffle_execution.png
```
Comment (aweisberg) [inline presto-docs/src/main/sphinx/admin/exchange-materialization.rst:36]: Capitalize first letter of bullet points?

"Schedule fewer partitions concurrently to reduce memory requirements"

File: presto-docs/src/main/sphinx/admin/exchange-materialization.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+``exchange_materialization_strategy``, ``partitioning_provider_catalog`` and ``hash_partition_count``:
+
+.. code-block:: sql
+
+    SET SESSION exchange_materialization_strategy='ALL';
+
+    -- Set partitioning_provider_catalog to the Hive connector catalog
+    SET SESSION partitioning_provider_catalog='hive';
+
+    -- We recommend to set hash_partition_count to be at least 5X-10X about the number of cluster size
```
Comment (aweisberg) [inline presto-docs/src/main/sphinx/admin/exchange-materialization.rst:52]:
```suggestion
    -- We recommend setting hash_partition_count to be at least 5X-10X cluster size
```

File: presto-docs/src/main/sphinx/admin/session-property-managers.rst [hunk 4]
```diff
@@ -62,6 +62,9 @@ Consider the following set of requirements:
 * All ETL queries (tagged with 'etl') are routed to subgroups under the ``global.pipeline`` group, and must be
   configured with certain properties to control writer behavior.
 
+* All high memory ETL queries (tagged with 'high_mem_etl') re routed to subgroups under the ``global.pipeline`` group,
```
Comment (aweisberg) [inline presto-docs/src/main/sphinx/admin/session-property-managers.rst:65]:
```suggestion
* All high memory ETL queries (tagged with 'high_mem_etl') are routed to subgroups under the ``global.pipeline`` group,
```

**Review by aweisberg — CHANGES_REQUESTED** [review]
I didn't check that the links all work, so assuming they are tested.

File: presto-docs/src/main/sphinx/admin/exchange-materialization.rst (hunk 1 above)
Comment (wenleix) [reply presto-docs/src/main/sphinx/admin/exchange-materialization.rst:32]: @aweisberg : What about `opens the opportunity for XXX`?

**Comment by wenleix** [comment]
@aweisberg : comments addressed

File: presto-docs/src/main/sphinx/admin/exchange-materialization.rst (hunk 1 above)
Comment (aweisberg) [reply presto-docs/src/main/sphinx/admin/exchange-materialization.rst]: That WFM

File: presto-docs/src/main/sphinx/admin/exchange-materialization.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+``exchange_materialization_strategy``, ``partitioning_provider_catalog`` and ``hash_partition_count``:
+
+.. code-block:: sql
+
+    SET SESSION exchange_materialization_strategy='ALL';
+
+    -- Set partitioning_provider_catalog to the Hive connector catalog
+    SET SESSION partitioning_provider_catalog='hive';
+
+    -- We recommend setting hash_partition_count to be at least 5X-10X about the number of cluster size
```
Comment (aweisberg) [inline presto-docs/src/main/sphinx/admin/exchange-materialization.rst:52]:
```suggestion
    -- We recommend setting hash_partition_count to be at least 5X-10X the cluster size
```

**Review by aweisberg — APPROVED** [review]
You missed part of one of the suggested changes, otherwise LGTM

**Pull request merged**